### PR TITLE
[RFC] vim-patch:7.4.606

### PR DESCRIPTION
**Pull request by Yamakaky**
```
Problem:    May crash when using a small window.
Solution:   Avoid dividing by zero. (Christian Brabandt)
```

https://github.com/vim/vim/commit/v7-4-606

Original patch:

``` diff
diff --git a/src/nvim/normal.c b/src/nvim/normal.c
--- a/src/nvim/normal.c
+++ b/src/nvim/normal.c
@@ -4457,6 +4457,8 @@ nv_screengo(oap, dir, dist)
     col_off2 = col_off1 - curwin_col_off2();
     width1 = W_WIDTH(curwin) - col_off1;
     width2 = W_WIDTH(curwin) - col_off2;
+    if (width2 == 0)
+   width2 = 1; /* avoid divide by zero */

 #ifdef FEAT_VERTSPLIT
     if (curwin->w_width != 0)
diff --git a/src/nvim/version.c b/src/nvim/version.c
--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -742,6 +742,8 @@ static char *(features[]) =
 static int included_patches[] =
 {   /* Add new patch number below this line */
 /**/
+    606,
+/**/
     605,
 /**/
     604,
```

### CODE REVIEW

File: src/nvim/normal.c [hunk 1]
```diff
@@ -3378,6 +3378,9 @@ static bool nv_screengo(oparg_T *oap, int dir, long dist)
   col_off2 = col_off1 - curwin_col_off2();
   width1 = curwin->w_width - col_off1;
   width2 = curwin->w_width - col_off2;
+  if (width2 == 0) {
+      width2 = 1; /* avoid divide by zero */
```
Comment (glts) [inline src/nvim/normal.c]: Style: two space indent, `//` comment style with capitalisation and punctuation (`// Avoid divide by zero.`), and as far as I know, put comments on a line of their own.

**Comment by glts** [comment]
Discussion about the original patch here https://groups.google.com/d/msg/vim_dev/lhyiH5MnBkw/eYXoyUyET8wJ Unfortunately no steps to reproduce. But the patch is simple enough, LGTM when the style issue is fixed. :+1: 

By the way, mark your pull requests [RFC] when ready to be reviewed.

**Comment by Yamakaky** [comment]
Done

**Comment by glts** [comment]
> Done

I'd say 1 out of 3 done, check my comment again :)

**Comment by Yamakaky** [comment]
Ouch, sorry ^^

**Comment by chrisbra** [comment]
On Di, 05 Mai 2015, David Bürgin wrote:

> Unfortunately no steps to reproduce. 

On very small windows when 'nu' is set, using gk might crash Vim with a 
floating point exception.

Best,
Christian
-- 
Die Zeit ist ein guter Arzt, aber ein schlechter Kosmetiker.
		-- William Somerset Maugham

File: src/nvim/normal.c (hunk 1 above)
Comment (Pyrohh) [inline src/nvim/normal.c]: @Yamakaky the indentation is still off.

**Comment by justinmk** [comment]
Merged, thanks @Yamakaky

**Pull request closed without merging**